libglusterfs/src/iobuf.c

```diff
@@ -570,7 +616,17 @@ iobuf_get_page_aligned(struct iobuf_pool *iobuf_pool, size_t page_size,
         req_size = iobuf_pool->default_page_size;
     }
 
-    iobuf = iobuf_get2(iobuf_pool, req_size + align_size);
+    req_size = req_size + align_size;
+    if (req_size <= USE_IOBUF_POOL_IF_SIZE_GREATER_THAN) {
+        iobuf = iobuf_get_from_small(req_size);
+        if (!iobuf)
+            gf_smsg(THIS->name, GF_LOG_WARNING, 0, LG_MSG_IOBUF_NOT_FOUND,
+                    NULL);
+        iobuf->ptr = GF_ALIGN_BUF(iobuf->ptr, align_size);
+        return iobuf;
```

Comment:
Do we really need this ? when `iobuf_get2()` is called, it will already check the size and choose the right option.

Reply:
done